packages/excalidraw/components/App.tsx

```diff
@@ -8735,13 +8822,21 @@ class App extends React.Component<AppProps, AppState> {
   ) => {
     file = await normalizeFile(file);
     try {
+      const elements = this.scene.getElementsIncludingDeleted();
       const ret = await loadSceneOrLibraryFromBlob(
         file,
         this.state,
-        this.scene.getElementsIncludingDeleted(),
+        elements,
         fileHandle,
       );
       if (ret.type === MIME_TYPES.excalidraw) {
+        // Restore the fractional indices by mutating elements and update the
+        // store snapshot, otherwise we would end up with duplicate indices
+        restoreFractionalIndices(elements.concat(ret.data.elements));
+        this.store.snapshot = this.store.snapshot.clone(
+          arrayToMap(elements),
+          this.state,
+        );
```

Comment:
Above updates all fractional indices in the snapshot as we purposefully never delete from it even though the elements were removed from the scene (so we could calculate the deltas and always go back). Not updating the indices could lead to us having duplicate indices in the scene, after a series of undo / redo commands. For details check `scene import via drag&drop should create new history entry`.

Reply:
Update, the above sync updates only invalid indices (likely in the dropped scene) due to improvements in #7693